server/channels/api4/channel_test.go

```diff
@@ -2026,8 +2026,7 @@ func TestGetChannelsForTeamForUser(t *testing.T) {
 		}
 
 		channels, resp, err = client.GetChannelsForTeamForUser(context.Background(), th.BasicTeam.Id, th.BasicUser.Id, false, resp.Etag)
-		// an error is expected as the server didn't return any data and the client still tries to unmarshal the response
-		require.Error(t, err)
+		require.NoError(t, err)
```

Comment:
>  I also wanted to ask for a test asserting that the bug that motivated this change is now gone.

This is the test that prompted the whole PR

Reply:
Ah, perfect! I was thinking of an explicit test covering exactly that, but if this is enough, I'm ok with it :)